client/shipping-label/state/actions.js

```diff
@@ -514,13 +515,18 @@ export const purchaseLabel = () => ( dispatch, getState, context ) => {
 					labelId: label.label_id,
 				} ) );
 				const state = getState().shippingLabel;
-				printDocument( getPrintURL( state.paperSize, labelsToPrint, context ) )
-					.then( () => {
+				const printUrl = getPrintURL( state.paperSize, labelsToPrint, context );
+				printDocument( printUrl, false )
+					.then( ( success ) => {
 						const noticeText = 1 === response.length
 								? __( 'Your shipping label was purchased successfully' )
 								: sprintf( __( 'Your %d shipping labels were purchased successfully' ), response.length );
 						dispatch( NoticeActions.successNotice( noticeText ) );
-						dispatch( exitPrintingFlow( true ) );
+						if ( success ) {
+							dispatch( exitPrintingFlow( true ) );
+						} else {
+							dispatch( { type: SHOW_PRINT_CONFIRMATION, printUrl } );
+						}
 					} )
 					.catch( ( err ) => {
 						console.error( err );
```

Comment:
At this point, it's better to just let the user refresh the page. We don't have logic in place to correctly recover from an error in this flow. For example, the EasyPost `shipment_id` is not reset, so another try at purchasing would give an EasyPost error for sure.